drop-transfer/src/check.rs

```diff
@@ -31,31 +31,43 @@ pub(crate) fn spawn(
 }
 
 async fn run(state: Arc<State>, xfer: Arc<IncomingTransfer>, logger: Logger) {
-    loop {
-        tokio::time::sleep(drop_config::ALIVE_CHECK_INTERVAL).await;
-
+    let mut remove_xfer = false;
+    for delay in drop_config::RETRY_INTERVALS {
         if !state.transfer_manager.is_incoming_alive(xfer.id()).await {
+            remove_xfer = true;
             break;
         }
 
         if make_request(&state, &xfer, &logger).await.is_break() {
+            remove_xfer = true;
             break;
         }
+
+        debug!(
+            logger,
+            "Client job will restart in {:?} seconds for {}",
+            delay,
+            xfer.id()
+        );
+
+        tokio::time::sleep(delay).await;
```

Comment:
Doesn't sleeping at the end of the loop in this case mean that the last duration is effectively useless?

Reply:
:+1: lemme fix it